weather_dl/download_pipeline/parsers_test.py

```diff
@@ -63,15 +63,177 @@ def test_throws_error(self):
+        with io.StringIO('{"section": {"key": "value", "list": "1.0/to/5.0/by/2.0"}}') as f:
+            actual = parse_config(f)
+            for key, val in actual['section'].items():
+                self.assertNotIn('\n', val)
+            self.assertListEqual(actual['section']['list'], ['1.0', '3.0', '5.0'])
+
+    def test_json_parses_mars_float_range_incremented_by_float(self):
+        with io.StringIO('{"section": {"key": "value", "list": "0.0/to/0.5/by/0.1"}}') as f:
+            actual = parse_config(f)
+            for key, val in actual['section'].items():
+                self.assertNotIn('\n', val)
+            self.assertEqual(actual['section']['list'], ['0.0', '0.1', '0.2', '0.30000000000000004', '0.4', '0.5'])
+
+    def test_json_parses_mars_date_range(self):
+        with io.StringIO('{"section": {"key": "value", "list": "2020-01-07/to/2020-01-09"}}') as f:
+            actual = parse_config(f)
+            for key, val in actual['section'].items():
+                self.assertNotIn('\n', val)
+            self.assertListEqual(actual['section']['list'], ['2020-01-07', '2020-01-08', '2020-01-09'])
+
+    def test_json_parses_mars_relative_date_range(self):
+        with io.StringIO('{"section": {"key": "value", "list": "-3/to/-1"}}') as f:
+            actual = parse_config(f)
+            for key, val in actual['section'].items():
+                self.assertNotIn('\n', val)
+
+            dates = [
+                datetime.date.today() + datetime.timedelta(-3),
+                datetime.date.today() + datetime.timedelta(-2),
+                datetime.date.today() + datetime.timedelta(-1),
+            ]
+
+            self.assertListEqual(actual['section']['list'], [d.strftime("%Y-%m-%d") for d in dates])
+
+    def test_json_parses_mars_date_range_incremented(self):
+        with io.StringIO('{"section": {"key": "value", "list": "2020-01-07/to/2020-01-12/by/2"}}') as f:
+            actual = parse_config(f)
+            for key, val in actual['section'].items():
+                self.assertNotIn('\n', val)
+            self.assertListEqual(actual['section']['list'], ['2020-01-07', '2020-01-09', '2020-01-11'])
+
+    def test_json_raises_syntax_error_missing_right(self):
+        with self.assertRaises(SyntaxError) as ctx:
+            with io.StringIO('{"section": {"key": "value", "list": "2020-01-07/to/"}}') as f:
+                parse_config(f)
+
+        self.assertEqual("Improper range syntax in '2020-01-07/to/'.", ctx.exception.args[0])
+
+    def test_json_raises_syntax_error_missing_left(self):
+        with self.assertRaises(SyntaxError) as ctx:
+            with io.StringIO('{"section": {"key": "value", "list": "/to/2020-01-07"}}') as f:
+                parse_config(f)
+
+        self.assertEqual("Improper range syntax in '/to/2020-01-07'.", ctx.exception.args[0])
+
+    def test_json_raises_syntax_error_missing_increment(self):
+        with self.assertRaises(SyntaxError) as ctx:
```

Comment:
Since I've provided the other test examples, I've found a better way to write these asserts: 

```
with self.assertRaisesRegex(SyntaxError, "Improper range syntax in '2020-01-07/to/2020-01-11/by/'."):
     # ...
```

Reply:
Modified it for all similar test cases in parset_test